### DISCUSSION TITLE
A function to places all files in current directory into folder based on there extension

### DISCUSSION THREAD

**Discussion by MANICX100** [fish-shell/fish-shell#9162] [Q&A] [answered]
In bash compatible shells I had the following function:
    
    orderfiles() {
    for filename in *; do
      if [[ -f "$filename" ]]; then
          base=${filename%.*}
          ext=${filename#$base.}
        mkdir -p "${ext}"
        mv "$filename" "${ext}"
      fi
    done
    }

To accomplish the same thing in fish my current non working example is:
  
    function orderfiles                                                           
        set list ./**.*                      
        for val in $list                       
                set ext  ( COMMAND to get file extension $val )   
                mkdir -p "$ext"               
                cp $val "./$ext" 
        end                                        
    end 

I have tried calling

      set base filename%.*
      set ext filename#$base.

elsewhere in my config and then ext in the function but that doesn't work

**Comment by faho** [comment, marked as answer]
You want `path extension -- $val`

See https://fishshell.com/docs/current/cmds/path.html#extension-subcommand

This will include the ".", to strip that off use `string replace . ""`:

```fish
set ext (path extension -- $val | string replace . "")
```

**Comment by MANICX100** [comment]
Thank you that works perfectly. :)
